server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -220,24 +220,16 @@ public static Query termsAndGenericFilter(Function function,
                                               boolean hasDocValues,
                                               IndexType indexType,
                                               Occur termQueryOccur) {
-        var flatUniqueValues = flattenUnique(values);
         var canUseArrayLengthIndex = context.tableInfo().versionCreated().onOrAfter(Version.V_5_9_0);
         BooleanQuery.Builder filterClauses = new BooleanQuery.Builder();
         Query genericFunctionFilter = genericFunctionFilter(function, context);
-        if (flatUniqueValues.isEmpty()) { // using flatUnique to catch nested empty arrays
+        if (values.isEmpty()) {
```

Comment:
`[[]]` is not empty so the execution should follow `else` path. It worked because the inner array is empty and we did index inner arrays' lengths. But the inner arrays' length fields will be removed by https://github.com/crate/crate/pull/16688.